src/app/profile/spaces/space.service.ts

```diff
@@ -0,0 +1,100 @@
+import { Injectable } from '@angular/core';
+import { Headers, Http } from '@angular/http';
+import { cloneDeep } from 'lodash';
+import { Logger } from '../../shared/logger.service';
+import { AuthenticationService } from '../../auth/authentication.service';
+import { Space } from '../../models/space';
+
+@Injectable()
+export class SpaceService {
+
+  private headers = new Headers({ 'Content-Type': 'application/json' });
+  private spacesUrl = process.env.API_URL + "spaces";
+  private nextLink: string = null;
+
+  private spaces: Space[] = [];
+  private spaceIdIndexMap = {};
+
+  constructor(private http: Http, private logger: Logger, private auth: AuthenticationService) {
+    if (this.auth.getToken() != null) {
+      this.headers.set('Authorization', 'Bearer ' + this.auth.getToken());
+    }
+    logger.log('SpaceService using url:' + this.spacesUrl)
+  }
+
+  getSpaces(pageSize: number = 20): Promise<Space[]> {
+    console.log("Is logger there in getspaces?" + (this.logger == null));
+    let url = this.spacesUrl + '?page[limit]=' + pageSize;
+
+    return this.http
+      .get(url, { headers: this.headers })
+      .toPromise()
+      .then(response => {
+        let spaces: Space[];
```

Comment:
You probably don't need to declare this up here, as you don't use it until you assign lower down.